modules/core/src/main/scala/com.snowplowanalytics.iglu/client/CirceClient.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2014-2021 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.iglu.client
+
+import cats.Monad
+import cats.data.EitherT
+import cats.effect.Clock
+import com.snowplowanalytics.iglu.client.resolver.registries.RegistryLookup
+import com.snowplowanalytics.iglu.client.resolver.{InitListCache, InitSchemaCache}
+import com.snowplowanalytics.iglu.client.validator.CirceValidator.WithCaching.{
+  InitValidatorCache,
+  SchemaEvaluationCache,
+  SchemaEvaluationKey,
+  SchemaEvaluationResult
+}
+import com.snowplowanalytics.iglu.core.SelfDescribingData
+import com.snowplowanalytics.lrumap.CreateLruMap
+import io.circe.{DecodingFailure, Json}
+
+final class CirceClient[F[_]] private (
+  resolver: Resolver[F],
+  schemaEvaluationCache: SchemaEvaluationCache[F]
+) {
+  def check(
+    instance: SelfDescribingData[Json] //TODO can we commit to Json here and get rid of type param?
+  )(implicit
+    M: Monad[F], //TODO can we move those to client constructor?
```

Comment:
The pattern we use almost everywhere in Snowplow is to put implicit typeclasses on the methods, not on the constructors.

I think this was mainly Anton's influence, although I cannot remember the details of his reasoning.  I remember an occasion where a code change was made more difficult because of implicits on a constructor, and I remember Anton saying something like "I told you so" (he phrased it better)... but I cannot remember any details and I cannot find a record of the conversation.

Generally, I have never found it a problem to have implicit typeclasses on the methods, so I'd prefer to stick with what's working for us.  Unless you have a strong reason to move them?